### PR TITLE
Add EM introducing constants, global variables, and EM Setting variables

### PR DESCRIPTION
**Pull request by ChemiKyle**
Adds the `Accessing Variables` exercise module

### CODE REVIEW

File: docs/guide_for_devs.md [hunk 1]
```diff
@@ -190,11 +190,113 @@ $module->sayHello();
...
 ```
 
 </details>
 <br />
+
+---
+
+### [Accessing Variables]({{ site.repo_root }}exercises/accessing_variables/)
+
+While working on this module, you will learn how to access constants and variables defined by REDCap. You will also cover using `project-settings` to allow users to set variables and accessing those variables.
```
Comment (pbchase) [inline docs/guide_for_devs.md]:
```suggestion
While working on this module, you will learn how to access constants and variables defined by REDCap. You will also use `project-settings` to allow set and access variables.
```

File: docs/guide_for_devs.md [hunk 2]
```diff
@@ -190,11 +190,113 @@ $module->sayHello();
...
+
+The goal of this exercise is to create a module that displays a user's:
+1. Username
+1. Admin (aka superuser) status
+1. Their user rights
+1. The current page path
+1. The current project's `project_id`
+1. The value of a variable set in the module's configuration menu
+
+Check the [official External Module documentation on functions](https://github.com/vanderbilt/redcap-external-modules/blob/testing/docs/framework/intro.md) for functions which will provide this information for you. Useful phrases to search for are "User" and "projectSetting".
```
Comment (pbchase) [inline docs/guide_for_devs.md]:
```suggestion
Check the [official External Module documentation on functions](https://github.com/vanderbilt/redcap-external-modules/blob/testing/docs/framework/intro.md) for functions which provide this information for you. Useful phrases to search for are "User" and "projectSetting".
```

File: docs/guide_for_devs.md [hunk 3]
```diff
@@ -190,11 +190,113 @@ $module->sayHello();
...
+
+Note that there is an unlisted property of the [`User` object](https://github.com/vanderbilt/redcap-external-modules/blob/testing/docs/framework/intro.md#user-method) that you will need: 
+
+```php
+$User = $this->framework->getUser();
+$User->username;
+```
+
+If your programming environment has a debugger available, it is highly recommended to use it to see the accessible constants. If you do not have a debugger or cannot connect it to your REDCap docker container, please view the list below. Note that most of these are available as functions.
```
Comment (pbchase) [inline docs/guide_for_devs.md]:
```suggestion
If your programming environment has a debugger available, you can use it to see the accessible constants. If you do not have a debugger or cannot connect it to your REDCap docker container, please view the list below. Note that most of these are available as functions.  
```

File: docs/guide_for_devs.md [hunk 4]
```diff
@@ -190,11 +190,113 @@ $module->sayHello();
...
+$User = $this->framework->getUser();
+$User->username;
+```
+
+If your programming environment has a debugger available, it is highly recommended to use it to see the accessible constants. If you do not have a debugger or cannot connect it to your REDCap docker container, please view the list below. Note that most of these are available as functions.  
+<details>
+<summary>Useful global constants
+</summary>
+
+- **PAGE**: The current loaded file and its path with the root being the active REDCap folder, (root being, i.e. `redcap_v9.3.5/`); this is the only constant listed that is not easily determined by a function
```
Comment (pbchase) [inline docs/guide_for_devs.md]:
```suggestion
- **PAGE**: The path portion of the URL for the current web page. This is the only constant listed that is not easily determined by a function.
```

File: docs/guide_for_devs.md [hunk 5]
```diff
@@ -190,11 +190,113 @@ $module->sayHello();
...
+</details>
+<br />
+
+As before, documentation _must_ be updated when introducing features. This is even more important when adding functionality that users interact with (e.g. an entry in `project-settings` in the `config.json`)!
+
+<details>
+<summary>Example Solution Via a Hook
+</summary>
+
+Note that this is a _bare minimum_ implementation; because the goal of this exercise is to find out how to access the variables needed, this solution only aims to show how to access those variables. When displaying information to users, you should _not_ use basic functions like `print_r` (unless the contents will be showing up in the body of a plugin page).
```
Comment (pbchase) [inline docs/guide_for_devs.md]:
```suggestion
This is a _bare minimum_ implementation that demonstrates how to access REDCap variables. The output is _ugly_. An attractive display of those variables is left as an exercise for the student.  
```

**Comment by pbchase** [comment]
Those are all the comments I have on the PR. I am sure Grammarly would sneer on my use of passive voice, but we're just going to roll with that for now.

File: docs/guide_for_devs.md (hunk 5 above)
Comment (ChemiKyle) [reply docs/guide_for_devs.md]: My zealous fervor for semicolons is balanced by the use of "left as an exercise to the student".

**Pull request merged**